GliaWidgets/Sources/AlertManager/AlertManager.swift

```diff
@@ -0,0 +1,140 @@
+    private var alertWindow: UIWindow?
+
+    init(
+        environment: Environment,
+        viewFactory: ViewFactory
+    ) {
+        self.environment = environment
+        self.viewFactory = viewFactory
+        composer = .init(
+            environment: .init(log: environment.log),
+            theme: viewFactory.theme
+        )
+    }
+}
+
+// MARK: - Public Methods
+extension AlertManager {
+    func present(
+        in placement: AlertPlacement,
+        as input: AlertInputType
+    ) {
+        guard input != currentAlert else { return }
+
+        let alertType = composer.composeAlert(
+            type: input,
+            in: placement
+        )
+
+        switch placement {
+        case .global:
+            presentGlobally(
+                type: alertType,
+                input: input
+            )
+        case let .root(viewController):
+            presentInRoot(
+                type: alertType,
+                input: input,
+                viewController: viewController
+            )
+        }
+    }
+
+    func overrideTheme(_ newTheme: Theme) {
+        self.composer.overrideTheme(with: newTheme)
+        self.viewFactory.overrideTheme(with: newTheme)
+    }
+}
+
+// MARK: - Private Methods
+private extension AlertManager {
+    func presentInRoot(
+        type: AlertType,
+        input: AlertInputType,
+        viewController: UIViewController
+    ) {
+        /// TODO - Will be done in a separate PR
+    }
+
+    func presentGlobally(
+        type: AlertType,
+        input: AlertInputType
+    ) {
+        let alertViewController = createAlertViewController(type: type)
+        alertViewController.onDismissed = { [weak self] in
+            self?.dismissGlobalAlert()
+        }
+        guard isAlertReplacable(offer: alertViewController) else { return }
+        self.alertViewController = alertViewController
+        self.currentAlert = input
+        createAlertWindow()
+        guard let alertWindow = alertWindow else { return }
+        alertWindow.isHidden = false
+        alertWindow.makeKeyAndVisible()
+        alertWindow.rootViewController?.present(
+            alertViewController,
+            animated: true,
+            completion: nil
+        )
+    }
+
+    func createAlertViewController(type: AlertType) -> AlertViewController {
+        .init(type: type, viewFactory: viewFactory)
+    }
+
+    func isAlertReplacable(offer: Replaceable) -> Bool {
+        guard let alertViewController else { return true }
+        switch alertViewController.isReplaceable(with: offer) {
+        case true:
+            return true
+        case false:
+            return false
+        }
+    }
+
+    func dismissGlobalAlert() {
+        alertViewController?.dismiss(animated: true)
+        alertViewController = nil
+        alertWindow?.isHidden = true
+        alertWindow = nil
+        currentAlert = nil
+    }
+}
+
+// MARK: - KeyWindow
+private extension AlertManager {
+    func createAlertWindow() {
```

Comment:
could elaborate a bit, for what purposes would we like to have a separate window for alerts?

Reply:
These are for mediaUprades and screensharing that need to be displayed in any view of the app, not limited to our Glia views

Reply:
See the implementation here https://github.com/salemove/ios-sdk-widgets/blob/master/GliaWidgets/Sources/OperatorRequestHandlerService/OperatorRequestHandlerService.swift